src/main/java/a00953080/comp3717/bcit/ca/newwesttriviaapp/activities/ResultActivity.java

```diff
@@ -28,26 +31,48 @@ protected void onCreate(Bundle savedInstanceState) {
 
         feedback      = (TextView)findViewById(R.id.feedBack);
         scoreView     = (TextView)findViewById(R.id.currentScore);
+        errorView     = (TextView)findViewById(R.id.error);
+        wagerText     = (EditText)findViewById(R.id.wager);
         nextQuestion  = (Button) findViewById(R.id.nextQuestionBtn);
 
-        score         = ((TriviaApp) getApplication()).getScore();
 
+        score         = ((TriviaApp) getApplication()).getScore();
         update();
     }
 
     public void update() {
         final String feedbackValue;
-
         feedbackValue = this.score.isPreviousAnswerCorrect() ? "Correct" : "Incorrect";
 
         scoreAnimation();
         changeBackGroundColor();
 
         feedback.setText(feedbackValue);
     }
+    public boolean validateWager(){
+        String value = wagerText.getText().toString();
+        if(value.isEmpty())
+            return true;
+        Long bet = Long.parseLong(value);
+        if(bet > this.score.getScore())
+            return false;
+        return true;
+    }
+    public void updateWager(){
+        String value = wagerText.getText().toString();
+        if(!value.isEmpty())
+            this.score.setWager(Long.parseLong(value));
+        else
+            this.score.setWager(0);
+    }
```

Comment:
I was thinking that maybe we should require the user to input a wager rather than setting a default one. What do you think about that?